.github/workflows/examples.yml

```diff
@@ -7,47 +6,54 @@ on:
       - main
   pull_request:
 
+env:
+  rust-version: 1.60.0
+  dfx-version: 0.11.1
+
 jobs:
   test:
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-        build: [ linux-stable ]
-        include:
-          - build: linux-stable
-            os: ubuntu-latest
-            rust: 1.60.0
-            dfx: 0.11.1
+        project-name:
+          [asset_storage, chess, counter, management_canister, print, profile]
 
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout repository
+        uses: actions/checkout@v3
 
-      - uses: actions/cache@v2
+      - name: Cache
+        uses: actions/cache@v3
         with:
           path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-            **/target
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-1
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            examples/${{ matrix.project-name }}/target/
+          key: ${{ runner.os }}-${{ matrix.project-name }}
```

Comment:
What is the rationale for not including the hash of the Cargo.lock files"?

Reply:
`cdk-rs` has `Cargo.lock` in `.gitignore`. So `${{ hashFiles('**/Cargo.lock') }}` will always be empty string.
I'm considering using `${{ hashFiles('**/Cargo.toml') }}` instead. Will have a try.

Reply:
for maximum speed, perhaps it might be worth considering to hash `cargo metadata --no-deps | jq ".packages[0].dependencies"`, that way a version bump won't cause new cache entry. 

`jq "[...].dependencies"` includes both `dependencies` and `dev-dependencies`

Reply:
tho now I think about it, the solution may not be bullet-proof, e.g. consider what happens when you bump `rustc` version in `rust-toolchain.toml`. 
For this reason, it may be a good idea to instead include `Cargo.toml` in git tree and use that.

Reply:
I haven't read through it yet, but you also might want to check out https://github.com/Swatinem/rust-cache

Reply:
Most crates in this repo are libraries which should better not to include Cargo.lock in version control.
The uses of `restore-keys` basically guarantees that there is always some cache to be restored. Producing new cache entries won't do much harm.
Including `rust-toolchain.toml` in the hash is a good idea. I will add it.